### PR TITLE
Added a contrib folder which contains Docker examples

### LINKED ISSUE #617: Docker Examples
I wanted to ask you, @dmacvicar, if you be interested in a couple docker examples? It might help people who want to try out your plugin and adds a easy of use for users. I also found it helpful when I need to manually build your plugin to test out new features, like the 0.12 release. 

Here are some examples below:

**Build plugin for Alpine images:**
```
# Building Libvirt Plugin Docker
FROM golang:alpine

# Update APK
RUN apk update && apk upgrade

# Install Dependencies
RUN apk add git make pkgconfig libvirt gcc libvirt-dev libc-dev

# Debugging packages
RUN apk add file nano

RUN go get -u golang.org/x/lint/golint
RUN go get github.com/dmacvicar/terraform-provider-libvirt

WORKDIR $GOPATH/src/github.com/dmacvicar/terraform-provider-libvirt
RUN make build
```

**Build plugin for Debian based images(Ubuntu, Debian, etc):**
```
# Building Libvirt Plugin Docker
FROM golang:stretch

# Update APK
RUN apt-get update -y && apt-get upgrade -y

# Install Dependencies
RUN apt-get install git make pkg-config libvirt0 gcc libvirt-dev libc-dev -y

# Debugging packages
RUN apt-get install file nano -y

RUN go get -u golang.org/x/lint/golint
RUN go get github.com/dmacvicar/terraform-provider-libvirt

WORKDIR $GOPATH/src/github.com/dmacvicar/terraform-provider-libvirt
RUN make build
``` 

**Run Terraform 0.12 and Libvirt plugin on Ubuntu 18.04:**
```
# Grab the Terraform binary
FROM hashicorp/terraform:0.12.5 AS terraform

# Grab the Libvirt plugin binary
FROM terraform-libvirt:latest AS libvirt

FROM ubuntu:18.04

# Set working dir
WORKDIR /root/

# Install Dependencies
RUN apt-get update -y

# Dependencies for Libvirt Plugin
RUN apt-get install dbus dmeventd dmsetup libapparmor1 libavahi-client3 libavahi-common-data libavahi-common3 libdbus-1-3 libdevmapper-event1.02.1 libdevmapper1.02.1 liblvm2app2.2 liblvm2cmd2.02 libnl-3-200 libnuma1 libreadline5 libvirt0 libyajl2 lvm2 readline-common -y

# Dependencies for Local Use(Optional)
RUN apt-get install wget git sshpass gettext python3 python3-pip xsltproc mkisofs -y

# Copy Terraform binary to image
COPY --from=terraform /bin/terraform /bin/

# Copy provider plugin to image(ONLY for x86, could adpat for multi-arch)
RUN mkdir -p /root/.terraform.d/plugins/linux_amd64/
COPY --from=libvirt /go/src/github.com/dmacvicar/terraform-provider-libvirt/terraform-provider-libvirt /root/.terraform.d/plugins/linux_amd64/

# ADD YOUR FILES HERE
COPY example.tf /root/

# Terraform commands
RUN terraform init

ENTRYPOINT /bin/bash
```

These are all pretty rough Dockerfiles, as I am only using them for quick testing, but I can clean them up and submit a PR if you are interested.

### PR DESCRIPTION
**Pull request by james-crowley**
@MalloZup and @dmacvicar when you get a chance could you look over the Docker examples I provided. 

Added a contrib folder which contains Docker examples for Alpine, Debian, openSUSE. Additionally, included a build Dockerfile for building the provider. Provided documentation for all Dockerfiles via README.md files.

Fixes #617

### CODE REVIEW

**Comment by MalloZup** [comment]
@james-crowley  I just had quick look on this, seems pretty good thx!

1) Make the commit message better then the actual ones.  `Fixes` , `Pushing changes` etc. Just squash them or make a good Title and description.

2) You can add a Link to the `contrib/Readme.md` in to the global `readme`. Imho you could reference it  here: 
https://github.com/dmacvicar/terraform-provider-libvirt#upstream-projects-using-terraform-libvirt

After that section, you can create one saying a short desc about it and the link

File: contrib/Alpine/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+## Build-Dependent Container Usage
+Build appropriate `Build` container, in this case it would be `Dockerfile_musl`:
+
+```cosnole
+docker build -f Dockerfile_musl -t provider-libvirt:v0.5.2-musl . --build-arg VERSION=v0.5.2
+```
+
+Now build the Main container:
```
Comment (MalloZup) [inline contrib/Alpine/README.md]: why `Main` is uppercased? I found this in all doc. Imho we can just use `main` lower :+1:

**Comment by james-crowley** [comment]
@MalloZup I updated the root's README. 

I tried to do a squash but something went terribly wrong as you can see with all my new commits. Any suggestions/fixes? Don't squash my commits that often.

**Comment by MalloZup** [comment]
@james-crowley  `git rebase` and `git sqaush` is a secret/dark art which Is difficult to explain. It is pita.

I don't mind if you close the PR and open new one. As you wish.
 IF you want to solve the conflicts, you need to `git pull upstream` then `rebase to upstream` and then you need to figure out the conflicts individually

**Comment by james-crowley** [comment]
I am going to close this, fix the issues then open a PR. Thanks for the insights into `git rebase`.

**Pull request closed without merging**